### DISCUSSION TITLE
Package Upgrade: 17.2 not higher than 17.1.4

### DISCUSSION THREAD

**Discussion by iambernics** [chocolatey/choco#2723] [General]
Hello,

"choco upgrade" doesn't update a package if the previous version contain three segment (17.1.4) and the new only two (17.2).
Why?

Regards,
Gabor

**Comment by pauby** [comment]
Can you give an example of the package you are trying this with? Can you also confirm which repository you are using?

**Reply by iambernics** [reply]
> Can you give an example of the package you are trying this with? Can you also confirm which repository you are using?

Self developed package from a private feed.
Choco.exe is able to connect to the feed but ignore the upgrade when the version increased from 17.1.4  to 17.2.
Works fine if the version changed to 17.2.0.

If this is the normal operation of the choco, I will skip the further investigation.

Content of nuspec

```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
  <metadata>
    <id>vs_2022_enterprise</id>
    <version>17.2.0</version>
    <owners>Microsoft</owners>
    <summary>Visual Studio 2022 Enterprise Edition</summary>
    <description>Visual Studio 2022 Enterprise Edition</description>
  </metadata>
</package>
```

**Reply by pauby** [reply]
This isn't normal for Chocolatey. But there was a very similar sounding issue that was raised an issue for on [cChoco](https://github.com/chocolatey/cChoco/issues/161) recently around Sonatype Nexus. Is this what you are using for your feed?

**Reply by iambernics** [reply]
> This isn't normal for Chocolatey. But there was a very similar sounding issue that was raised an issue for on [cChoco](https://github.com/chocolatey/cChoco/issues/161) recently around Sonatype Nexus. Is this what you are using for your feed?

No, JFrog Artifactory.
